Applications/PixelPaint/Filters/GenericConvolutionFilter.cpp

```diff
@@ -49,8 +49,12 @@ GenericConvolutionFilter<N>::~GenericConvolutionFilter()
 }
 
 template<size_t N>
-void GenericConvolutionFilter<N>::apply(const Filter::Parameters& parameters)
+void GenericConvolutionFilter<N>::apply(const OwnPtr<Filter::Parameters>& maybe_parameters)
 {
+    if (!maybe_parameters)
+        return;
```

Comment:
This API is weird. How about we just don't call `apply` when there are no parameters instead? :)

Reply:
and add more `if`s?
sure, why not.